### ISSUE TITLE
nvgShapeAntiAlias doesn't allow for single pixel stroke widths

### ISSUE THREAD

**Issue by xeekworx** [memononen/nanovg#471]
If I disable shape anti-alias with nvgShapeAntiAlias any lines I draw with a stroke width of 1.0 will not be visible at all. With shape anti-alias ON the line shows just fine, though quite wider than 1 pixel due to the aggressive anti-aliasing.

For me this results in no line drawn:
```cpp
void renderer_nanovg::on_draw_line(const int32_t x1, const int32_t y1, const int32_t x2, const int32_t y2, const color& clr) const
{
    if (m_nvgcontext) {
        NVGcontext * vg = reinterpret_cast<NVGcontext *>(m_nvgcontext);

        nvgShapeAntiAlias(vg, 0);
        nvgBeginPath(vg);
        nvgMoveTo(vg, (float)x1, (float)y1);
        nvgLineTo(vg, (float)x2, (float)y2);
        nvgStrokeColor(vg, nvgRGBA(clr.r, clr.g, clr.b, clr.a));
        nvgStrokeWidth(vg, 1.f);
        nvgStroke(vg);
        nvgShapeAntiAlias(vg, 1);
    }
}
```

**Comment by mkeida** [comment]
Having same issue :/

**Comment by memononen** [comment]
If you increase the line width, at what point it starts to be visible?
Crisp 1px antialiased lines should be possible.
Can you post a screenshot of the issues?

**Comment by memononen** [comment]
I fixed the line rendering when AA is disable. The problem I saw was not what you explained, though. The line was just rendered a bit thinner (it was still visible, though). Can you test the latest and see if that fixed your problem.

I would still be interested to see the too wide line.

**Comment by mkeida** [comment]
Thanks a lot! I appreciate your hard work! Line started to be visible at 2f width ... Top and left borders were drawn correctly with 2 pixels and bottom and right cornes were drawn only with one pixel... With 1f width bottom and right borders of rectangle weren't visible at all... Top and left borders were drawed correctly with 1 pixel... I will post screenshots later when I will be home. I'm using LWJGL and its Java binding of NanoVG so i will probably need to rework my project to c++ to use this.

**Comment by memononen** [comment]
I can see @mkeida 's message on my email, but not here. One thing to remember when drawing lines, and especially 1px lines is how sampling works.

This article explains it very well:
http://www.mobtowers.com/html5-canvas-crisp-lines-every-time/

This can cause lines to appear dim and fat, or disappear when drawing them without AA.

**Comment by mkeida** [comment]
Yeah sorry for that I deleted my comment by mistake. Everything works fine now, atleast for me. Thanks a lot for your time and hard work!

**Comment by memononen** [comment]
@xeekworx both.
